docs/conceptual/runtime/contracts/contracts_module.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Contracts Module
+---
+
+The [SRML Contracts module](https://substrate.dev/rustdocs/master/srml_contracts/index.html)
+provides the ability for the runtime to deploy and execute [WebAssembly
+(Wasm)](https://webassembly.org/) smart contracts.
+
+## Wasm Engine
+
+The SRML Contracts module depends on a Wasm sandboxing interface defining the Wasm execution engine
+available within the runtime. This is currently implemented with
+[`wasmi`](https://github.com/paritytech/wasmi), a Wasm interpreter.
+
+## Features
+
+The SRML Contracts module has a number of familiar and new features for the deployment and execution
+of smart contracts.
+
+### Account Based
+
+The Contracts module uses an account-based system similar to many existing smart contract platforms.
+To the Substrate runtime, contract accounts are just like normal user accounts; however, in addition
+to an `AccountID` and `Balance` that normal accounts have, a contract account also has associated
+contract code and some persistent contract storage.
+
+### Two Step Deployment
+
+Deploying a contract with the Contracts module takes two steps:
+
+1. Store the Wasm contract on the blockchain.
+2. Instantiate a new account, with new storage, associated with that Wasm contract.
+
+This means that multiple contract instances, with different constructor arguments, can be
+initialized using the same Wasm code, reducing the amount of storage space needed by the Contracts
+module on your blockchain.
+
+### Runtime Environment Types
+
+For writing contracts and interacting with the runtime, a set of types are available (e.g.
+`AccountId`, `Balance`, `Hash`, `Moment`). These types can be user defined for custom runtimes, or
+the supplied defaults can be used.
+
+### Contract Calls
+
+Calls to contracts can alter the storage of the contract, create new contracts, and call other
+contracts. Because Substrate provides you with the ability to write custom runtime modules, the
+Contracts module also enables you to make asynchronous calls directly to those runtime functions on
+behalf of the contract's account.
+
+### Sandboxed
+
+The Contracts module is intended to be used by any user on a public network. This means that
+contracts only have the ability to directly modify their own storage. To provide safety to the
+underlying blockchain state, the Contracts module enables revertible transactions, which roll back
+any changes to the storage by contract calls that do not complete successfully.
+
```

Comment:
```suggestion

While using a sandboxed environment is desirable from a security perspective, we could be looking to read data from the smart contract's storage on the runtime or vice versa. This is a feature that exists on Substrate. The smart contract's write storage access will not be allowed form the runtime level but it will give us access to read it.
```

Reply:
Maybe @pepyakin or @ascjones could elaborate more.